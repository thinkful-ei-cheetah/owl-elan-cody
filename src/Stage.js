import React from 'react';

function Stage({store}) {
    const participantOnStage = store.map(item => item.onStage ?
        <div className="stage" key = {item.id}>
        <img src={item.avatar} alt = "avatar" />
        {item.name}
        </div>: null);
        
    return (
        <div className="stage-container">
            {participantOnStage}
        </div>
    )
   
}
export default Stage