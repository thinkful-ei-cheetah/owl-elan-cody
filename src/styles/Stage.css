#root {
    display: flex;
}

.stage-container {
    display: flex;
    flex-flow: row;
    justify-content: space-around;
    width: 80%;
    background-color: slategrey;
}

.stage {
    text-align: center;
}

.stage > img {
    max-width: 128px;
}