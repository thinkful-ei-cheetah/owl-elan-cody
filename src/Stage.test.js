import React from 'react';
import ReactDOM from 'react-dom';
import Stage from './Stage'
import participants from './participants';
import renderer from 'react-test-renderer';

describe('Stage Participants', () => {
    it('renders without crashing' , () => {
        const div = document.createElement('div');
        ReactDOM.render(<Stage store={participants} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders UI as expected', () => {
        const tree = renderer
            .create(<Stage store={participants} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
});