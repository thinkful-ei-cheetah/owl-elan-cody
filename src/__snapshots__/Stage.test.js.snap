// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage Participants renders UI as expected 1`] = `
<div
  className="stage-container"
>
  <div
    className="stage"
  >
    <img
      alt="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    Koren Templeton
  </div>
  <div
    className="stage"
  >
    <img
      alt="avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    Ashla Attwool
  </div>
</div>
`;
